src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -332,13 +336,36 @@ public function denormalize($data, string $type, string $format = null, array $c
             }
 
             $value = $this->validateAndDenormalize($resolvedClass, $attribute, $value, $format, $context);
+
+            if ($value instanceof DenormalizationResult) {
+                if (!$value->isSucessful()) {
+                    $invariantViolations += $value->getInvariantViolationsNestedIn($attribute);
+
+                    continue;
+                }
+
+                $value = $value->getDenormalizedValue();
+            }
+
             try {
                 $this->setAttributeValue($object, $attribute, $value, $format, $context);
             } catch (InvalidArgumentException $e) {
                 throw new NotNormalizableValueException(sprintf('Failed to denormalize attribute "%s" value for class "%s": '.$e->getMessage(), $attribute, $type), $e->getCode(), $e);
             }
         }
 
+        if ($context[self::COLLECT_INVARIANT_VIOLATIONS] ?? false) {
+            if (!empty($extraAttributes)) {
+                $invariantViolations[] = new InvariantViolation(sprintf('Extra attributes are not allowed ("%s" are unknown).', implode('", "', $extraAttributes)));
```

Comment:
Message to the wrong argument ;)
```suggestion
                $invariantViolations[] = new InvariantViolation(null, sprintf('Extra attributes are not allowed ("%s" are unknown).', implode('", "', $extraAttributes)));
```

Reply:
Fixed 😅

Reply:
`$data` will be the value of the last attribute, which might not even be one the extra ones.
I think you could either pass null to have one message listing the extra attributes or loop over the extra attributes and add  one message for each of them with `$denormalizedData[$attribute]` as value.